aws/resource_aws_flow_log_test.go

```diff
@@ -406,3 +438,49 @@ resource "aws_flow_log" "test" {
 }
 `, rName, rName, rName)
 }
+func testAccFlowLogConfig_LogFormat(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_vpc" "test" {
+  cidr_block = "10.0.0.0/16"
+
+  tags = {
+    Name = %q
+  }
+}
+
+resource "aws_iam_role" "test" {
+  name = %q
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": [
+          "ec2.amazonaws.com"
+        ]
+      },
+      "Action": [
+        "sts:AssumeRole"
+      ]
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_cloudwatch_log_group" "test" {
+  name = %q
+}
+
+resource "aws_flow_log" "test" {
+  iam_role_arn   = "${aws_iam_role.test.arn}"
+  log_group_name = "${aws_cloudwatch_log_group.test.name}"
+  traffic_type   = "ALL"
+  vpc_id         = "${aws_vpc.test.id}"
+  log_format     = "${version} ${vpc-id} ${subnet-id}"
```

Comment:
Unfortunately the log format syntax collides with Terraform string interpolation syntax, which generates configuration errors:

```
--- FAIL: TestAccAWSFlowLog_LogFormat (1.86s)
    testing.go:569: Step 0 error: config is invalid: 3 problems:

        - Invalid reference: A reference to a resource type must be followed by at least one attribute access, specifying the resource name.
        - Invalid reference: A reference to a resource type must be followed by at least one attribute access, specifying the resource name.
        - Invalid reference: A reference to a resource type must be followed by at least one attribute access, specifying the resource name.
```

Can you please escape the dollar signs and verify this works as expected? Thank you.

```suggestion
  log_format     = "$${version} $${vpc-id} $${subnet-id}"
```

Reply:
done.